src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -189,13 +189,13 @@ func (s *fileSystemSource) shardAvailability(
 	var tr xtime.Ranges
 	for i := 0; i < len(readInfoFilesResults); i++ {
 		result := readInfoFilesResults[i]
-		if result.Err.Error() != nil {
-			s.log.WithFields(
-				xlog.NewField("shard", shard),
-				xlog.NewField("namespace", namespace.String()),
-				xlog.NewField("error", result.Err.Error()),
-				xlog.NewField("targetRangesForShard", targetRangesForShard),
-				xlog.NewField("filepath", result.Err.Filepath()),
+		if err := result.Err.Error(); err != nil {
+			s.log.With(
+				zap.Uint32("shard", shard),
+				zap.Stringer("namespace", namespace),
+				zap.Error(err),
```

Comment:
nit: you've been putting `zap.Error()` at the end. Maybe just stick with that.

Reply:
The two are isomorphic, and I prefer how this reads right now. For edits - I like you're able to add/modify fields a lot easier because they're a line at a time and all together. In other instances where we're doing it at the end, we don't have an associated `With(..)` so it we have to do it that way. 

Going to leave as is unless you feel strongly.

Reply:
actually, reworking to not use a `With()` (unless we reuse the logger) at all will save the extra logger's alloc. And address your concern. Will rework

Reply:
done